src/script/interpreter.cpp

```diff
@@ -1847,16 +1847,14 @@ static bool ExecuteWitnessScript(const Span<const valtype>& stack_span, const CS
     return true;
 }
 
-static bool VerifyTaprootCommitment(const std::vector<unsigned char>& control, const std::vector<unsigned char>& program, const CScript& script, uint256& tapleaf_hash)
+uint256 ComputeTapleafHash(uint8_t leaf_version, const CScript& script)
```

Comment:
I mildly object the the use of the type `CScript` here.  While it is true that BIP-341 names this stack item "script", this value is only interpreted as a CScript in the case of BIP-342 where the `leaf_version` is equal to `TAPROOT_LEAF_TAPSCRIPT` (modulo `TAPROOT_LEAF_MASK`).  Otherwise it is simply a bytestring.